pkg/scheduler/cache/node_tree.go

```diff
@@ -59,7 +59,8 @@ func (na *nodeArray) next() (nodeName string, exhausted bool) {
 	return nodeName, false
 }
 
-func newNodeTree(nodes []*v1.Node) *NodeTree {
```

Comment:
Why don't we move node_tree.go to internal/cache?

Reply:
@misterikkit said:
There are some types in the current cache package, namely NodeInfo, which should remain where they are. These are actually used outside of the scheduler, and need to be decoupled from the cache implementation.

Reply:
Yes, but NodeTree is a recent addition. I would be surprised if it's used outside of the scheduler.

Reply:
You are right, NodeTree is suitable for moving to internal/cache.I'll do it.